### PR TITLE
IoK/FB/051719 - Offer new downsample_2d function to perform 2x2 bilinear

### PR DESCRIPTION
**Pull request by ikatsavounidisFB**
filtering instead of decimation

- Controlled by #define DECIMATION (line 4954, EbPictureAnalysisProcess.c)
- Minor bug fixing in original decimation_2d
- Fix compiler warning in cfl_avx2.c
- Add XCode project generator script

### CODE REVIEW

File: Source/Lib/Common/Codec/EbPictureAnalysisProcess.c [hunk 1]
```diff
@@ -174,20 +174,55 @@ void decimation_2d(
...
+    uint32_t horizontal_index;
+    uint32_t vertical_index;
+    uint32_t input_stripe_stride = input_stride * decim_step;
+    uint32_t decim_horizontal_index;
+    const uint32_t half_decim_step = decim_step >> 1;
+
+    for (input_samples += half_decim_step * input_stride, vertical_index = half_decim_step; vertical_index < input_area_height; vertical_index += decim_step) {
+        uint8_t *prev_input_line = input_samples - input_stride;
+        for (horizontal_index = half_decim_step, decim_horizontal_index = 0; horizontal_index < input_area_width; horizontal_index += decim_step, decim_horizontal_index++) {
+            uint32_t sum = (uint32_t) prev_input_line[horizontal_index - 1] + (uint32_t) prev_input_line[horizontal_index] + (uint32_t) input_samples[horizontal_index - 1] + (uint32_t) input_samples[horizontal_index];
```
Comment (kylophone) [inline Source/Lib/Common/Codec/EbPictureAnalysisProcess.c]: Casting is not necessary since `sum` is `uint32_t`.

**Comment by hassount** [comment]
@ikatsavounidisFB 
Thank you for your contribution. We have done some BD Rate tests over the AOM test set and have noticed some gains and losses due to these changes: 
For example:

- In the blue_sky_360p_60f  clip we are noticing a 2% gain in the context of M1 and 3.6% gain in the context of M8. 

- In the STARCRAFT_60f_420 clip we are noticing a 25% loss in M1 and 30% loss in M8. 

If we can tune this further to box where the gains / losses are coming from that would allow us to take advantage of where this feature is helping and leave aside where it's not.

Clips: https://media.xiph.org/video/derf/objective-1-fast.tar.gz

**Comment by joelsole** [comment]
Could you add how to use the XCode project generator script to the README?
I didn't manage to make it work.

**Comment by hassount** [comment]
Closing as #353 has been merged

**Pull request closed without merging**